src/api/routes/faucets.ts

```diff
@@ -148,6 +144,10 @@ export function createFaucetRouter(db: PgWriteStore): express.Router {
   const FAUCET_STACKING_WINDOW = 2 * 24 * 60 * 60 * 1000; // 2 days
   const FAUCET_STACKING_TRIGGER_COUNT = 1;
 
+  const STX_FAUCET_NETWORKS = getStxFaucetNetworks();
+  const STX_FAUCET_KEYS = (process.env.FAUCET_PRIVATE_KEY ?? testnetKeys[0].secretKey).split(',');
+  let stxKeyIndex = 0;
```

Comment:
I think this needs to be stateless, so the loop just tries sending from all accounts when getting a TooMuchChaining

Reply:
Just so I understand: You mean always try with the first one and upon failure retry with all others at the same time? Wouldn't that send more STX than necessary in case all other txs succeed?

Reply:
Start on first one, upon failure go to next one, upon success then break the loop. Does that make sense?

Reply:
Done